src/transport/SessionManager.cpp

```diff
@@ -304,15 +304,40 @@ CHIP_ERROR SessionManager::PrepareMessage(const SessionHandle & sessionHandle, P
         snprintf(ackBuf, sizeof(ackBuf), " (Ack:" ChipLogFormatMessageCounter ")", payloadHeader.GetAckMessageCounter().Value());
     }
 
+    char addressStr[Transport::PeerAddress::kMaxToStringSize] = { 0 };
+    switch (sessionHandle->GetSessionType())
+    {
+    case Transport::Session::SessionType::kGroupOutgoing: {
+        // We already checked above that we have a fabric entry for our fabric
+        // index.
+        auto * groupSession   = sessionHandle->AsOutgoingGroupSession();
+        auto * fabricInfo     = mFabricTable->FindFabricWithIndex(fabricIndex);
+        auto multicastAddress = Transport::PeerAddress::Multicast(fabricInfo->GetFabricId(), groupSession->GetGroupId());
```

Comment:
Please deal with fabricInfo being null (not sure it can happen, but since this layer crosses and also runs async on retries, perhaps this can fail if there are pending retransmits for a fabric after a fabric removal).

Reply:
This code is not hit for retransmits.  

In the `kGroupOutgoing` case, which is where we are here, the switch earlier in the function has:

```
        const FabricInfo * fabric = mFabricTable->FindFabricWithIndex(groupSession->GetFabricIndex());
        VerifyOrReturnError(fabric != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
```

so unless the fabric table changes during the execution of this function we can't get null here...  That's what the coment about "We already checked above" is trying to communicate.  I will make it clearer that it was handled in the switch above.

Reply:
Actually, I will just reuse that code here.  That will be even clearer.